public/pages/CreateTransform/containers/CreateTransformForm/CreateTransformForm.test.tsx

```diff
@@ -0,0 +1,408 @@
+                          <div style={{ padding: "25px 25px" }}>
+                            <CreateTransformForm
+                              {...props}
+                              transformService={services.transformService}
+                              rollupService={services.rollupService}
+                              indexService={services.indexService}
+                              core={coreServicesMock}
+                            />
+                          </div>
+                        )}
+                      />
+                      <Route path={ROUTES.TRANSFORMS} render={(props) => <div>Testing transform landing page</div>} />
+                      <Redirect from="/" to={ROUTES.CREATE_TRANSFORM} />
+                    </Switch>
+                  </ModalProvider>
+                )
+              }
+            </ServicesConsumer>
+          </ServicesContext.Provider>
+        </CoreServicesContext.Provider>
+      </Router>
+    ),
+  };
+}
+
+describe("<CreateTransformForm /> spec", () => {
+  browserServicesMock.transformService.getMappings = jest.fn().mockResolvedValue({
+    ok: true,
+    response: sampleMapping,
+  });
+
+  browserServicesMock.rollupService.getMappings = jest.fn().mockResolvedValue({
+    ok: true,
+    response: sampleMapping,
+  });
+
+  browserServicesMock.transformService.searchSampleData = jest.fn().mockResolvedValue({
+    ok: true,
+    response: {
+      data: indexData,
+      total: { value: 1 }
+    }
+  });
+
+  it("renders the component", async () => {
+    const { container } = renderCreateTransformFormWithRouter();
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it("set breadcrumbs when mounting", async () => {
+    renderCreateTransformFormWithRouter();
+
+    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledTimes(4);
+    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledWith([
+      BREADCRUMBS.INDEX_MANAGEMENT,
+      BREADCRUMBS.TRANSFORMS,
+      BREADCRUMBS.CREATE_TRANSFORM,
+    ]);
+  });
+
+  it("routes back to transform landing page if cancelled", async () => {
+    const { getByTestId, getByText } = renderCreateTransformFormWithRouter();
+
+    expect(getByTestId("createTransformCancelButton")).toBeEnabled();
+
+    userEvent.click(getByTestId("createTransformCancelButton"));
+
+    await waitFor(() => getByText("Testing transform landing page"));
+  })
+
+  it("does not move to step 2 without info", async () => {
+    const { getByTestId, getByText, getByLabelText, queryByText } = renderCreateTransformFormWithRouter();
+    browserServicesMock.transformService.getTransform = jest.fn().mockResolvedValue({
+      ok: false,
+      response: {},
+    });
+
+    expect(getByTestId("createTransformNextButton")).toBeEnabled();
+
+    userEvent.click(getByTestId("createTransformNextButton"));
+    await waitFor(() => {},{timeout:2000});
+
+    // Currently no pop up warnings?
```

Comment:
I think there might be some inline error text that we can check. But knowing that it is still on step 1 is also fine.